### ISSUE TITLE
could you use the maven proxies for the download?

### ISSUE THREAD

**Issue by drdamour** [takari/takari-maven-plugin#25]
i have proxies setup in my settings.xml, can those be used when downloading stuff?

**Comment by gzm55** [comment]
use  [system property](https://docs.oracle.com/javase/6/docs/technotes/guides/net/properties.html).

**Comment by drdamour** [comment]
That’s what I did, but not what I’m asking. Since maven has proxy as a first class concept why can’t this utilize that

**Comment by mosabua** [comment]
Because the wrapper runs BEFORE maven ever gets into the picture.

**Comment by gzm55** [comment]
This plugin runs in maven, and should already using the proxy settings of maven.
Whereas, the maven wrapper runs before and without maven, so it is too heavy to mimic what maven does.

**Comment by drdamour** [comment]
even when i run `mvn -N io.takari:maven:0.7.6:wrapper` it can't use the maven proxy to download stuff?

**Comment by gzm55** [comment]
cause plugin [shares](https://github.com/takari/takari-maven-plugin/blob/master/src/main/java/io/takari/maven/plugins/WrapperMojo.java#L76) the downloader with wrapper

**Comment by mosabua** [comment]
For that action it does use the proxy..

**Comment by drdamour** [comment]
i don't believe that it did...the download went direct and failed i had to use http.proxy to make ti work.  was a few months back let me try again

**Comment by drdamour** [comment]
indeed `mvn -N io.takari:maven:0.7.6:wrapper` is using the proxy from settings.xml for most things, but when it gets to download the gz file it doesn't use the proxy.  Since maven is involved at this point could it use the proxy?

**Comment by mosabua** [comment]
That command does not download the gz file. It only downloads the wrapper jar.

**Comment by drdamour** [comment]
it definitely tries to dl the gz file, I see it appearing from that command

**Comment by gzm55** [comment]
imo, `mvn -N io.takari:maven:0.7.6:wrapper` is a rare command only involved  when add/update the wrapper version, and should be executed manually from dev machine. So switching to another proxy method just in these cases is not a problem.

The wrapper is downloaded directly from a `URL`, not a `group:artifact:version`. It's not very straightly to reuse the infra of maven, instead, one need also fetch the proxy settings from maven objects, and inject into downloader.

**Comment by drdamour** [comment]
Ok, but it’s weird most the things it’ll dl (the stuff through maven dependency tree) respects the proxy but the gz download doesn’t...so it’s like some stuff works but other stuff doesn’t.  Even though it’s not common it is fundamental and the first thing someone has to do. Often devs set a proxy in settings and forget it’s even there.  So first step is pretty confusing to us stuck behind zscaler and the like which is becoming more and more common.

**Comment by gzm55** [comment]
try `MVNW_REPOURL` pointing to a non-proxy downloadable uri.

**Comment by drdamour** [comment]
try that to what end?

**Comment by gzm55** [comment]
plugin will download the wrapper from
```
$MVNW_REPOURL/io/takari/maven-wrapper/$ver/maven-wrapper-$ver.tar.gz
```

so let MVNW_REPOURL point to you internal private mirror behind the firewalls, then devs can successfully download w/o proxy.

**Comment by mosabua** [comment]
In a corporation this is what you would do since you should be running an internal Maven repository manager there..

**Comment by drdamour** [comment]
Except when it’s saas like azure DevOps and still goes through the zscaler...or should we not be doing that to. 

Anyways seems like no one thinks the download of the gz should use the same proxy as the other things when initializing a repo so...thanks

**Comment by gzm55** [comment]
this plugin already [supports `<mirrors>`](https://github.com/takari/takari-maven-plugin/blob/2c1506d7c2f26c0b151e6880f5105a004ea20c81/src/main/java/io/takari/maven/plugins/WrapperMojo.java#L137). With the similar method, try post a pr for your environment.
